libgadget/metal_return.c

```diff
@@ -1021,3 +1021,435 @@ stellar_density(const ActiveParticles * act, MyFloat * StellarAges, MyFloat * Ma
 
     return priv->VolumeSPH;
 }
+
+
+/* Here comes the neighbour finding. We want to return metals to the nearest N neighbours to the star.
+ * As a seed for the Hsml we find the kth nearest neighbour algorithm using a heap.
+ * This is always an underestimate of hsml, but should be enough for a gradient lookup to be efficient.*/
+#define NUMNB 50
+
+struct Neighbour
+{
+    int part;
```

Comment:
This is undefined if the neighbour is from a remote rank?

Reply:
I think in practice I don't need this part variable at all - I was keeping hold of it so that I could do metal return by just weighting the k nearest neighbours, but I decided I didn't want to monkey with the big run like that.